### PR TITLE
feat: implement job category

### PR DESCRIPTION
**Pull request by idax6797**
> [!NOTE]  
> This has also altered the naming of the service to manager, and set up the files in the same structure as done with the other managers.

This pull request introduces a new feature for managing job categories.
This includes a new controller, service, and repository for job categories.

### New Feature: Job Categories

* **JobCategoryController**: Added a new `JobCategoryController` class to handle API requests for job categories, including methods for pagination and HATEOAS links. (`src/CitMovie.Api/JobCategoryController.cs`)
* **JobCategoryService**: Introduced a `JobCategoryService` class to manage business logic for job categories. (`src/CitMovie.Business/JobCategoryService.cs`)
* **JobCategoryRepository**: Added a `JobCategoryRepository` and its interface to handle data access for job categories. (`src/CitMovie.Data/JobCategoryRepository/IJobCategoryRepository.cs`, `src/CitMovie.Data/JobCategoryRepository/JobCategoryRepository.cs`) [[1]](diffhunk://#diff-32436a6fcfbeccffa032188a4a7dc23676831d9eff5626cfc860cf50a0cc95e0R1-R10) [[2]](diffhunk://#diff-7a635da3bad3d4852adea6b30efcd243450667e1fc17185c28e0cc43d2d5e2f2R1-R33)
* **JobCategoryDto**: Created a `JobCategoryDto` class for data transfer objects related to job categories. (`src/CitMovie.Models/DTOs/JobCategoryDto.cs`)

### CODE REVIEW

File: src/CitMovie.Data/JobCategoryRepository/JobCategoryRepository.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    public class JobCategoryRepository : IJobCategoryRepository
+    {
+        private readonly DataContext _context;
+
+        public JobCategoryRepository(DataContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<IList<JobCategoryDto>> GetAllJobCategoriesAsync(int page, int pageSize)
```
Comment (MarLaslo) [inline src/CitMovie.Data/JobCategoryRepository/JobCategoryRepository.cs:15]: To my understanding here we return the Domain object and not dto.  In the manager we do the conversion

File: src/CitMovie.Business/Managers/JobCategoryManager.cs
Comment (MarLaslo) [inline src/CitMovie.Business/Managers/JobCategoryManager.cs:1]: Also I think we should have an IJobCategoryManager interface as we have for all the other managers.

**Review by MarLaslo — CHANGES_REQUESTED** [review]

File: src/CitMovie.Data/JobCategoryRepository/JobCategoryRepository.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        {
+            _context = context;
+        }
+
+        public async Task<IList<JobCategory>> GetAllJobCategoriesAsync(int page, int pageSize)
+        {
+            return await _context.JobCategories
+                .Skip(page * pageSize)
+                .Take(pageSize)
+                .Select(jc => new JobCategory
```
Comment (MarLaslo) [inline src/CitMovie.Data/JobCategoryRepository/JobCategoryRepository.cs:20]: Also here not neccessary probably

**Review by MarLaslo — APPROVED** [review]

File: src/CitMovie.Data/JobCategoryRepository/JobCategoryRepository.cs [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        private readonly DataContext _context;
+
+        public JobCategoryRepository(DataContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<IEnumerable<JobCategory>> GetAllJobCategoriesAsync(int page, int pageSize)
+        {
+            return await _context.JobCategories
```
Comment (JuliusKryger) [inline src/CitMovie.Data/JobCategoryRepository/JobCategoryRepository.cs:17]: Here kind of the same as last PR, we do not expect to update these only read them, so we can optimize this by letting EF know that with:

```suggestion
            return await _context.JobCategories
            .AsNoTracking()
```

**Review by JuliusKryger — CHANGES_REQUESTED** [review]

**Pull request merged**